src/history/index.js

```diff
@@ -12,24 +12,55 @@ const snapshotRecorder = new Recorder({ path: SNAPSHOTS_PATH, fileExtension: 'ht
 const versionRecorder = new Recorder({ path: VERSIONS_PATH, fileExtension: 'md' });
 
 export async function recordSnapshot(serviceId, documentType, content) {
-  return snapshotRecorder.record({
+  const isFirstRecord = !await snapshotRecorder.isTracked(serviceId, documentType);
+  const prefix = isFirstRecord ? 'Start tracking' : 'Update';
+  const changelog = `${prefix} ${serviceId} ${documentType}`;
+
+  const recordResult = await snapshotRecorder.record({
     serviceId,
     documentType,
-    content
+    content,
+    changelog
   });
+
+  return {
+    ...recordResult,
+    isFirstRecord
+  };
 }
 
 export async function recordVersion(serviceId, documentType, content, snapshotId) {
+  return _recordVersion({ serviceId, documentType, content, snapshotId });
+}
+
+export async function recordRefilter(serviceId, documentType, content, snapshotId) {
+  return _recordVersion({ serviceId, documentType, content, snapshotId, isRefiltering: true });
+}
+
+async function _recordVersion({ serviceId, documentType, content, snapshotId, isRefiltering }) {
   if (!snapshotId) {
     throw new Error(`A snapshot ID is required to ensure data consistency for ${serviceId}'s ${documentType}`);
   }
 
-  return versionRecorder.record({
+  const isFirstRecord = !await versionRecorder.isTracked(serviceId, documentType);
+  let prefix = isFirstRecord ? 'Start tracking' : 'Update';
+  prefix = isRefiltering ? 'Refilter' : prefix;
+
+  let changelog = `${prefix} ${serviceId} ${documentType}`;
+
+  changelog += `\n\nThis version was recorded after filtering snapshot ${config.get('history.publish') ? config.get('history.snapshotsBaseUrl') : ''}${snapshotId}`;
+
+  const recordResult = await versionRecorder.record({
     serviceId,
     documentType,
     content,
-    details: `This version was recorded after filtering snapshot ${config.get('history.publish') ? config.get('history.snapshotsBaseUrl') : ''}${snapshotId}`
+    changelog
   });
+
+  return {
+    ...recordResult,
+    isFirstRecord
```

Comment:
> Note: the only reason `isFirstRecord` is exposed to the caller at this time is for logging in `src/index.js`. This should be refactored later.

Reply:
It is also used for event emission, thus to not notify when it is a first record.